exercises/hamming/README.md

```diff
@@ -31,9 +31,10 @@ The Hamming distance between these two DNA strands is 7.
 
 # Implementation notes
 
-The Hamming distance is only defined for sequences of equal length. This means
-that based on the definition, each language could deal with getting sequences
-of equal length differently.
+The Hamming distance is only defined for sequences of equal length, so
+an attempt to calculate it between sequences of different lengths should
+not work. The general handling of this situation (e.g., raising an
+exception vs returning a special value) may differ between languages.
```

Comment:
Can you update this to the implementation in this track? We `throw and error` in this case: https://github.com/exercism/javascript/blob/master/exercises/hamming/hamming.spec.js#L34

Reply:
It appears like the testsuite may be behind the testdata.  The testdata has more than one case where an exception my be raised.  
see:
https://github.com/exercism/problem-specifications/blob/master/exercises/hamming/canonical-data.json#L135
and
https://github.com/exercism/problem-specifications/blob/master/exercises/hamming/canonical-data.json#L144

Reply:
Ok let me quickly do the test-case updation in another PR and you go ahead and update the README according to canonical implementation in this PR. I just did not want the `may differ between languages` part to remain there.

Reply:
Updated: https://github.com/exercism/javascript/pull/458